### ISSUE TITLE
maven-deploy and maven-install bash and batch scripts should terminate on mvn command error

### ISSUE THREAD

**Issue by perlun** [jOOQ/jOOQ#12942]
### Expected behavior
`maven-deploy.sh` should not continue executing if one of the `mvn` command invocations fail

### Actual behavior
Errors are silently ignored and the script happily carries on, even though the `mvn deploy:deploy-file` command has failed.

### Steps to reproduce the problem

```shell
$ ./maven-deploy.sh -u https://artifacts.example.com/repository/some-repository-name -r artifacts
```

Quoting myself from https://github.com/jOOQ/jOOQ/issues/12939#issuecomment-1021064947:

> On another note; would it be worth adding `set -e` (and potentially `-o pipefail`) to this script as well? Some people hate this kind of error handling (and it surely has some drawbacks), but... as the script currently stands, if the upload fails with e.g. `401 Unauthorized` or `403 Forbidden`, the script will happily carry on and try to upload all the other files as well.
>
> I personally like the idea of "fail fast" and I think it could be applied here as well. Failing to upload one of the files could just as well error with some reasonable error message instead of just continuing with the rest of the files.

You _could_ even consider adding `set -u` (fail on undefined variables) but it's even more controversial and will likely require script changes. But at the very least, `set -e -o pipefail` would be worth considering.

Note that `set -e` is considered bad practice by some people; this link might be of interest: e.g. http://mywiki.wooledge.org/BashFAQ/105.

Another option would be to do the `||` approach instead, i.e. run `mvn deploy:deploy-file -Dfile=jOOQ-pom/pom.xml                                     -DgroupId=org.jooq.trial-java-8 -DartifactId=jooq-parent                    -DrepositoryId=$REPOSITORY -Durl=$URL -Dversion=$VERSION -Dpackaging=pom || handle_error` and define `handle_error` as a shell function instead (which in turn could print an error and `exit 1`). That's "safer" in one sense but causes a bit of noise in the actual script code.

Anyway, those are implementation details and I think the first question is: _are we happy about the current semantics_ or do we want to fail more gracefully in case of e.g. 401/403 being returned from the Maven repository?

### Versions

- jOOQ: 3.16.2
- Java: 11.0.13
- OS: Debian GNU/Linux 11 (`bullseye`)

**Comment by lukaseder** [comment]
Thanks a lot for your report. Definitely worth investigating. I don't have much of an opinion here yet regarding the right way to handle errors, but you're probably right about aborting the deployment in case of a 401 or 403.

Will look into this

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.2 (https://github.com/jOOQ/jOOQ/issues/13777), 3.16.8 (https://github.com/jOOQ/jOOQ/issues/13778), and 3.15.12 (https://github.com/jOOQ/jOOQ/issues/13779)

**Comment by perlun** [comment]
> For the `bat` scripts, `|| exit /b` will do the trick on each command.

Interesting! I wasn't aware that Windows/DOS batch files supported this syntax as well. :slightly_smiling_face: :+1:
